egcg_core/clarity.py

```
@@ -295,9 +293,21 @@ def get_sample_release_date(sample_id):
     if not s:
         return None
     procs = connection().get_processes(type='Data Release EG 1.0', inputartifactlimsid=s.artifact.id)
-    if not procs:
-        return None
-    elif len(procs) != 1:
+    if len(procs) == 1:
+        return procs[0].date_run
+    else:
         app_logger.warning('%s Processes found for sample %s with Artifact %s', len(procs), sample_id, s.artifact.id)
+
+
+def get_library_id(sample_id):
+    if not get_sample(sample_id):
         return None
-    return procs[0].date_run
+    artifacts = connection().get_artifacts(
+        sample_name=sample_id,
+        type='Analyte',
+        process_type='AUTOMATED - Clean Up ALP'
+    )
+    if len(artifacts) == 1:
+        return artifacts[0].udf.get('Raw Library ID')
+    else:
+        app_logger.warning("%s 'Clean Up ALP' Artifacts found for sample %s", (len(artifacts), sample_id))
```

Could you raise an exception instead of a warning